test/functional/5-reload-doorhanger.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-env node, mocha */
+/* eslint-disable no-unreachable */
+
+// for unhandled promise rejection debugging
+process.on("unhandledRejection", r => console.error(r)); // eslint-disable-line no-console
+
+const {assert} = require("chai");
+const utils = require("./utils");
+const firefox = require("selenium-webdriver/firefox");
+const Context = firefox.Context;
+const webdriver = require("selenium-webdriver");
+const By = webdriver.By;
+const DELAY = process.env.DELAY ? parseInt(process.env.DELAY) : 500;
+
+describe("reload survey doorhanger", function() {
+  // This gives Firefox time to start, and us a bit longer during some of the tests.
+  this.timeout(DELAY * 15);
+
+  let driver;
+
+  // runs ONCE
+  before(async () => {
+    driver = await utils.setupWebdriver.promiseSetupDriver(
+      utils.FIREFOX_PREFERENCES,
+    );
+    await utils.setupWebdriver.installAddon(driver);
+    await driver.sleep(DELAY);
+  });
+
+  after(() => {
+    driver.quit();
+  });
+
+  describe("shows a survey after reloading the page 6 times max", function() {
+    let studyPings;
+    let tries = 0;
+
+    async function checkDoorhangerPresent() {
+      driver.setContext(Context.CHROME);
+      const result = await driver.findElements(By.id("cookie-restriction-notification"));
+      return !!result.length;
+    }
+
+    before(async () => {
+      await utils.setPreference(driver, "privacy.trackingprotection.enabled", true);
+      await driver.sleep(DELAY);
+
+      const time = Date.now();
+      driver.setContext(Context.CONTENT);
+      await driver.get("https://itisatrap.org/firefox/its-a-tracker.html");
```

Comment:
Should we also test that we show the door-hanger on a page without trackers?

Reply:
sure!